Add StreetView Tool (wo/ Google layers)

chrismayer opened the pull request:
This ports the StreetView tool (see #71), showing the SV panorama for the clicked position on the map. 

This does not show any Google layers on the map so far.

weskamm commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+    onToggle: function (btn, pressed) {
+        var me = this;
+
+        // detect the overlay layer group (to add psoition layer)
```
typo

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+        window.gm_authFailure = function() {
+            reactOnMissingGmapsApi();
+            return;
+        };
```
Can this be simplified to 

```javascript
window.gm_authFailure = reactOnMissingGmapsApi;
```

?

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+                        anchorYUnits: 'pixels',
+                        src: view.vectorIcon
+                    }))
+                })
```
Would be nice if only the style was configurable.

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+        me.streetViewService.getPanoramaByLocation(latLng, 50, function (data, status) {
+            if (status === google.maps.StreetViewStatus.OK) {
+                var win = me.streetViewWin;
+                var title = view.svWinTitlePrefix + ' Image Date: ' + data.imageDate;
```
Probably best to make the text configurable? Probably also for the way you detect the Overlay group by name, above somewhere.

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+     */
+    handlePovChanged: function () {
+        var me = this;
+        me.updatePositionFeature();
```
Aliasing unneeded here IMO. Also below.

chrismayer replied:
I suggest to leave this here beacuse of backward compatibility to the old version and more important
because it gives us the envelope for further things to be done on the certain events in the future.

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+
+        // normalize and convert to radiant
+        heading = me.normaliseDegrees(heading);
+        heading = heading / 180 * Math.PI;
```
I think there is a BasiGX util doing this... unsure though.

marcjansen replied:
I was thinking of theses meathods here: http://terrestris.github.io/BasiGX/master/docs/#!/api/BasiGX.view.button.Measure, but they do not help you exactly

marcjansen commented on app/controller/button/StreetViewTool.js:
```diff
@@ -0,0 +1,380 @@
+                    me.svPanorama, 'pov_changed', me.handlePovChanged.bind(me));
+                google.maps.event.addListener(
+                    me.svPanorama, 'position_changed', me.handlePositionChanged.bind(me));
+
```
Will we need to unregister these if the relevant components get destroyed?

geographika commented on app/view/button/StreetViewTool.js:
```diff
@@ -0,0 +1,96 @@
+    xtype: 'cmv_streetview_tool',
+
+    requires: [
+        'CpsiMapview.controller.button.StreetViewToolController'
```
Should this be CpsiMapview.controller.button.StreetViewTool (based on the filename of the controller)?

chrismayer replied:
Correct, thanks.

chrismayer commented:
Thanks everybody for your reviews. I addressed your comments (if not commented otherwise) and I additionally introduced an alert if no SV panorama is available at the clicked position.
